### PR TITLE
feat: sign_up_user_flow_improvement

### PR DESCRIPTION
**Pull request by xunlutas**
add deleteUserByEmail api

### CODE REVIEW

File: src/users/user.service.ts [hunk 1]
```diff
@@ -121,4 +120,16 @@ export class UserService {
     const foundUser = await this.userModel.find({ email: emailDto.email }).exec();
     return foundUser.length > 0;
   }
+
+  /**
+   * delete user by email
+   * @param emailDto
+   */
+
+  async deleteUserByEmail(emailDto: CheckEmailDto): Promise<{ message?: string }> {
```
Comment (tatoMa) [inline src/users/user.service.ts:129]: What is the difference between this deleteUserByEmail service you wrote and removeUserById one we already have? 

Are you sure we are truly going to delete a user but not using soft delete?

File: src/users/user.controller.ts [hunk 2]
```diff
@@ -14,6 +14,10 @@ export class UserController {
   async checkEmail(@Body() emailDto: CheckEmailDto): Promise<boolean> {
     return await this.userService.checkEmail(emailDto);
   }
+  @Post("delete")
```
Comment (tatoMa) [inline src/users/user.controller.ts:17]: Any reason why not to use @Delete DI?

File: src/users/user.service.ts (hunk 1 above)
Comment (xunlutas) [reply src/users/user.service.ts:129]: I am not sure which one is better. I think the data is still in the database if using soft delete. So some users who didn't registered successfully still can login directly.

File: src/users/user.service.ts [hunk 3]
```diff
@@ -115,10 +114,16 @@ export class UserService {
...
+   * Check if user is deleted and active by email
    * @param emailDto
    */
   async checkEmail(emailDto: CheckEmailDto): Promise<boolean> {
     const foundUser = await this.userModel.find({ email: emailDto.email }).exec();
-    return foundUser.length > 0;
+    if (foundUser.length > 0) {
+      const user = await this.userModel.findOne({ email: emailDto.email }).exec();
+      const { isActivated, isDeleted } = user;
+      return !isActivated || isDeleted ? false : true;
```
Comment (tatoMa) [inline src/users/user.service.ts:125]: This logic is a little confusing. If you need to reverse the result to false : true, you can simply add a ! (exclamation) before all your logic and put the existing logic into a bracket.

Comment (xunlutas) [reply src/users/user.service.ts:125]: fixed it

**Review by tatoMa — APPROVED** [review]
LGTM

**Review by zhuocun — APPROVED** [review]
Great job dude!

**Pull request merged**